plugin/pulpcore/plugin/changeset/model.py

```diff
@@ -91,17 +95,17 @@ def update(self, model):
             model (pulpcore.plugin.Content): A fetched content model object.
         """
         self.model = model
-        known = {a.model.relative_path: a for a in self.artifacts}
+        known = {a.content_artifact.relative_path: a for a in self.artifacts}
         self.artifacts.clear()
-        for artifact in model.artifacts.all():
+        for content_artifact in ContentArtifact.objects.filter(content=model):
```

Comment:
Unfortunate the django does not support query through the explicit thru table :/